server/datasources.json

```diff
@@ -0,0 +1,10 @@
+{
+  "memory": {
```

Comment:
Can we take this PR as an opportunity to switch to intention/purpose based names instead of implementation based names?